.github/workflows/kind-e2e.yaml

```diff
@@ -202,98 +187,12 @@ jobs:
-            endpoint = ["https://mirror.gcr.io"]
-
-        # This is needed in order to support projected volumes with service account tokens.
-        # See: https://kubernetes.slack.com/archives/CEKK1KTN2/p1600268272383600
-        kubeadmConfigPatches:
-          - |
-            kind: ClusterConfiguration
-            metadata:
-              name: config
-            apiServer:
-              extraArgs:
-                "service-account-issuer": "kubernetes.default.svc"
-                "service-account-signing-key-file": "/etc/kubernetes/pki/sa.key"
-            networking:
-              dnsDomain: "${CLUSTER_DOMAIN}"
-
-        nodes:
-        - role: control-plane
-          image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
-          extraMounts:
-          - containerPath: /var/lib/etcd
-            hostPath: /tmp/etcd
-        - role: worker
-          image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
-        - role: worker
-          image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
-        - role: worker
-          image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
-        - role: worker
-          image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
-        EOF
-
-        kind create cluster --config kind.yaml --wait 5m
-
-    - name: Install metallb
-      shell: bash
-      run: |
-        # Install metallb
-        # disable the webhook-mode because there is a bug:https://github.com/metallb/metallb/issues/1597
-        # this webhook only checks crd, it has not effects to out e2e tests
-        curl https://raw.githubusercontent.com/metallb/metallb/v0.13.5/config/manifests/metallb-native.yaml -k | \
-         sed '0,/args:/s//args:\n        - --webhook-mode=disabled/' | \
-         sed '/apiVersion: admissionregistration/,$d' | \
-         kubectl apply -f -
-
-        # Add Layer 2 config
-        network=$(docker network inspect kind -f "{{(index .IPAM.Config 0).Subnet}}" | cut -d '.' -f1,2)
-        cat <<EOF | kubectl apply -f -
-        apiVersion: metallb.io/v1beta1
-        kind: IPAddressPool
-        metadata:
-          name: first-pool
-          namespace: metallb-system
-        spec:
-          addresses:
-          - $network.255.1-$network.255.250
-        ---
-        apiVersion: metallb.io/v1beta1
-        kind: L2Advertisement
-        metadata:
-          name: example
-          namespace: metallb-system
-        EOF
-
     - name: Setup local registry
       run: |
+       # In the build job above , while setting up the registry, volume is set up. But when registry is set up in chainguard-dev/actions, it does not have a volume, so docker stop and rm is added below
+       docker stop $REGISTRY_NAME
+       docker rm $REGISTRY_NAME
+
         # Run a registry.
         docker run -d --restart=always \
           -v $HOME/artifacts/registry:/var/lib/registry \
```

Comment:
I upstreamed a change to chainguard-dev/actions to accept a volume for the registry - https://github.com/chainguard-dev/actions/pull/246

We can now set the property on the setup-kind step and we shouldn't need this anymore
```yaml
   registry-volume: $HOME/artifacts/registry
```